### PR TITLE
Hide BBC placeholder blocks when an image has loaded

### LINKED ISSUE #6429: Images with transparencies show the placeholder through the transparent bits
**Is your feature request related to a problem? Please describe.**
If a journalist adds a PNG with transparency in it to a CPS STY, then Simorgh displays the image with the BBC blocks image placeholder visible through it (on both Canonical and AMP).

Screen Shot 2020-05-06 at 10 53 46

[Asset where the image above lives.](https://www.bbc.co.uk/news/uk-politics-49560557)

**Describe the solution you'd like**
If the image has a transparency, then the placeholder should not be visible, and the background should match the background of the page.

I suspect there will need to be some accessibility considerations around ensuring the background colour is correct in various reading modes.

**Describe alternatives you've considered**
N/A

**Testing notes**
- Create unit tests to check if placeholder present when a image is passed in.
- Alt test matches the image, rather than placeholder.
- Check colour matches the page background.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by 12**
Resolves #6429.

**Overall change:**
Remove the BBC blocks placeholder when an image fires the `onload` event, to prevent the BBC blocks being visible through a transparent image.

**Code changes:**
- Use a React state hook to determine whether to show BBC blocks placeholder
- If hook is `true`, apply some additional CSS using the [styled-components CSS prop](https://styled-components.com/docs/api#css-prop).
- Update unit tests

**Notes:**
- Image fade is preserved
- A user may see a split second of nothing whilst the image is rendered in the DOM, but usually only on slow devices
---
- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ImageWithPlaceholder/index.jsx [hunk 1]
```diff
@@ -34,12 +39,15 @@ const ImageWithPlaceholder = ({
...
+  const ImageParent = isLoaded ? ImageWithoutPlaceholder : ImagePlaceholder;
   const imageProps = { alt, src, sizes, width, srcset, fade };
-  const imageToRender = <Image {...imageProps} />;
+  const imageToRender = (
+    <Image onLoad={() => setIsLoaded(true)} {...imageProps} />
+  );
 
   return (
-    <ImagePlaceholder ratio={ratio}>
+    <ImageParent ratio={ratio}>
```
Comment (12) [inline src/app/containers/ImageWithPlaceholder/index.jsx:50]: ## Other Solutions Assessed
**1. Using Styled Components CSS Prop**
```js
<ImagePlaceholder css={hidePlaceholder && `background: none`}>
...
</ImagePlaceholder>
```

Nice and tidy. Though it causes a lot of snapshot updates, due to Styled Components internal workings (I assume). Not a bad thing.

**2. Nesting `ImagePlaceholder` inside another SC that has some background logic**
```js
const StyledPlaceholder = styled(ImagePlaceholder)`
  background: ${({ isLoaded }) => isLoaded && 'none'};
`;
```

Adds an extra class as above, therefore same issue with snapshots.


This solution doesn't make any changes to snapshots, and still works as expected.

**Review by FK78 — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]
👍

**Review by simonsinclair — APPROVED** [review]
I love this approach, Rhys. Nice work.

**Review by andrewscfc — APPROVED** [review]

**Pull request merged**